Sources/tart/OCI/Layerizer/DiskV2.swift

```diff
@@ -121,11 +121,14 @@ class DiskV2: Disk {
         // Launch a fetching and decompression task
         group.addTask {
           // No need to fetch and decompress anything if we've already done so
-          if try pullResumed && Digest.hash(diskURL, offset: diskWritingOffset, size: uncompressedLayerSize) == uncompressedLayerContentDigest {
-            // Update the progress
-            progress.completedUnitCount += Int64(diskLayer.size)
+          if pullResumed {
```

Comment:
I'm not sure if we need this change, because currently [short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation) will kick in if `pullResumed` is `false`, and hash won't be calculated.

Reply:
Not according to the profiling I did. Both parts of the expression are getting evaluated regardless.

Reply:
Is it possible that `pullResumed` was set to `true` when the profiling was made?

Reply:
Unlikely, I was running pulls to completion between profiles.

Reply:
I've just done some debugging and I think that we can safely revert this because the compiler is correct w.r.t. this piece of code.

It's just that there's one more call to the `Digest.hash()` that is visible in the Instruments "CPU Profiler", and it is located below:

https://github.com/cirruslabs/tart/blob/33b5cfe2ed4b74e35ab60eefc53aede25b54db18/Sources/tart/OCI/Layerizer/DiskV2.swift#L143

Reply:
On Sequoia it's still the case. Maybe something to do about the `try`, don't know.